include/mls/credential.h

```diff
@@ -38,6 +39,20 @@ struct BasicCredential
   static const CredentialType type;
 };
 
+struct X509Credential
+{
+  X509Credential() {}
+
+  explicit X509Credential(const std::vector<bytes>& der_chain);
+
+  SignaturePublicKey public_key;
+  std::vector<hpke::Certificate> chain;
+
+  TLS_SERIALIZABLE(public_key)
```

Comment:
This is wrong.  You need to write a custom serializer that concatenates the chain and then marshals it as a `tls::vector<2>`.  And the reverse in the deserialize direction.